js-packages/@prestojs/rest/src/Paginator.ts

```diff
@@ -17,14 +17,29 @@ export default class Paginator<State, InternalState> implements PaginatorInterfa
     setCurrentState: (set: State) => void;
     setInternalState: (set: InternalState) => void;
 
+    constructor(currentStatePair = null, internalStatePair = null) {
+        if ((currentStatePair || internalStatePair) && !(currentStatePair && internalStatePair)) {
+            throw new Error(
+                'If one of `currentStatePair` and `internalStatePair` are specified both must be'
+            );
+        }
+        if (currentStatePair && internalStatePair) {
+            this.replaceStateControllers(currentStatePair, internalStatePair);
+        }
+    }
+
     /**
-     * Paginator received 2 tuples of a state and state setter pair. This is expected to
+     * Paginator receives 2 tuples of a state and state setter pair. This is expected to
      * match the same interface as `useState` in React. The following is a valid simple usage:
      *
      * ```js
-     * new Paginator(useState(), useState());
+     * const paginator = new Paginator(useState(), useState());
```

Comment:
See comments above re internal vs URL state. Should add a link to wherever it is documented how to use the URL for state